tests/cpp/test_multidevice_pipeline.cpp

```diff
@@ -239,7 +236,7 @@ INSTANTIATE_TEST_SUITE_P(
         testing::Values(true),
         testing::Values(false),
         testing::Values(false),
-        testing::Values(0),
+        testing::Values(0, 1),
```

Comment:
Could you also add different non-outermost resharding in this test https://github.com/NVIDIA/Fuser/blob/9c2359fe3f7655a0e4726a06ba186ecb58a5c2e4/tests/cpp/test_multidevice_resharding.cpp#L230C1-L230C46 ?

This test doesn't run the multidevice fusion (we should probably change that btw) but checks many more scenarios at the symbolic level. If `isLowerableToCommunication` is updated as I suggest in another comment, this test should spot if there are issues (possibly with `replaceValInAllExprInputsAndFusionOutputs`)